core/src/test/java/com/crawljax/core/CandidateElementExtractorTest.java

```diff
@@ -173,15 +175,34 @@ public void whenFollowExternalUrlDoFollow() throws IOException, URISyntaxExcepti
 		CrawljaxConfiguration config = builder.build();
 		CandidateElementExtractor extractor = newElementExtractor(config);
 
-		List<CandidateElement> extract = extractFromTestFile(extractor);
+		List<CandidateElement> extract = extractFromTestFile(extractor, defaultFile);
 
 		assertThat(config.getCrawlRules().followExternalLinks(), is(true));
 		assertThat(extract, hasSize(3));
 	}
 
-	private List<CandidateElement> extractFromTestFile(CandidateElementExtractor extractor) throws URISyntaxException {
+	@Test
+	public void testExtractShouldIgnoreDownloadFiles() throws Exception {
+
+		CrawljaxConfigurationBuilder builder = CrawljaxConfiguration.builderFor("http://example.com");
+		builder.crawlRules().click("a");
+		CrawljaxConfiguration config = builder.build();
+
+		CandidateElementExtractor extractor = newElementExtractor(config);
+
+		String file = "/site/element-extractor/index.html";
```

Comment:
The file could be moved to `/candidateElementExtractorTest/`.

Reply:
Got it!
I have added a param in the `extractFromTestFile` method. That can set the self-definition file path.